### ISSUE TITLE
LSTM layer occasionally outputs junk

### ISSUE THREAD

**Issue by dgcrouse** [soumith/cudnn.torch#242]
I am using ngimel's fork of torch-rnn to add cuDNN support. Every once in a while when first starting training, my loss function returns NaN, presumably from the LSTM layer (the only one from cuDNN) outputting junk. When I stop and restart it begins returning normal values (initial losses are in the 3-4 range). Is this a bug with cuDNN or something else? I am running CUDA 8 RC and cuDNN 5.0 on a GTX 960.

**Comment by soumith** [comment]
There's several cudnn 5.0 version. Can you specify the exact one?

From torch you can print cudnn.version

**Comment by dgcrouse** [comment]
Version is 5005

**Comment by dgcrouse** [comment]
Error appears to have been in my code, not CUDNN, as it appears I would occasionally pass junk data to the layer. Fixing my code appeared to stop the NaNs.
